frontend/apps/cloudstorage/cloudstorage.lua

```diff
@@ -0,0 +1,381 @@
+    }
+        self.cloud_dialog = ButtonDialogTitle:new{
+            title = _("Choose cloud storage type"),
+            title_align = "center",
+            buttons = buttons,
+    }
+
+    UIManager:show(self.cloud_dialog)
+    return true
+end
+
+function CloudStorage:openCloudServer(url)
+    local tbl
+    if self.type == "dropbox" then
+        tbl = DropBox:run(url, self.password)
+    elseif self.type == "ftp" then
+        tbl = Ftp:run(self.address, self.username, self.password, url)
+    end
+    if tbl and #tbl > 0 then
+        self:swithItemTable(url, tbl)
+        return true
+    elseif not tbl then
+        UIManager:show(InfoMessage:new{
+            text = _("Cannot fetch list folder!\nCheck configuration or network connection."),
+            timeout = 3,
+        })
+        table.remove(self.paths)
+        return false
+    else
+        UIManager:show(InfoMessage:new{text = _("Empty folder") })
+        return false
+    end
+end
+
+function CloudStorage:onMenuSelect(item)
+    if item.callback then
+        if item.url ~= nil then
+            table.insert(self.paths, {
+                url = item.url,
+            })
+        end
+        item.callback()
+    elseif item.type == "file" then
+        self:downloadFile(item)
+    else
+        table.insert(self.paths, {
+            url = item.url,
+        })
+        if not self:openCloudServer(item.url) then
+            table.remove(self.paths)
+        end
+    end
+    return true
+end
+
+function CloudStorage:downloadFile(item)
+    local lastdir = G_reader_settings:readSetting("lastdir")
+    local cs_settings = self:readSettings()
+    local download_dir = cs_settings:readSetting("download_dir") or lastdir
+    local path = download_dir .. '/' .. item.text
+    if lfs.attributes(path) then
+        UIManager:show(ConfirmBox:new{
+            text = _("File exist! Would you like to override it?"),
+            ok_callback = function()
+                self:cloudFile(item, path)
+            end
+        })
+    else
+        self:cloudFile(item, path)
+    end
+end
+
+function CloudStorage:cloudFile(item, path)
+    local path_dir = path
+    local buttons = {
+        {
+            {
+                text = _("Download file"),
+                callback = function()
+                    --Dropbox
+                    if self.type == "dropbox" then
+                        local callback_close = function()
+                            self:onClose()
+                        end
+                        UIManager:scheduleIn(1, function()
+                            DropBox:downloadFile(item, self.password, path_dir, callback_close)
+                        end)
+                        UIManager:close(self.download_dialog)
+                        UIManager:show(InfoMessage:new{
+                            text = _("Downloading may take several minutes..."),
+                            timeout = 1,
+                        })
+                    -- FTP
```

Comment:
nitpick: redundant comment, the logic in the code already clearly indicates it's handling FTP. Same for the dropbox comment above.